presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -856,10 +867,18 @@ protected Object visitNotExpression(NotExpression node, Object context)
         @Override
         protected Object visitLogicalBinaryExpression(LogicalBinaryExpression node, Object context)
         {
+            LogicalBinaryExpression.Type type = node.getType();
             Object left = process(node.getLeft(), context);
+            if (isPredicate && type == LogicalBinaryExpression.Type.AND && Boolean.FALSE.equals(left)) {
```

Comment:
why the distinction between a predicate and not a predicate is important?

Reply:
I was thinking about null values, but it does not matter. Removed.